diff -Nur -x '*.orig' -x '*.rej' e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/util.pm mezzanine_patched_e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/util.pm
--- e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/util.pm	2008-01-20 22:46:34.000000000 +0000
+++ mezzanine_patched_e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/util.pm	2008-01-20 22:44:06.000000000 +0000
@@ -741,8 +741,8 @@
           && die "Failed to set Samba password for account $username.\n";
         exit 0;
     }
-
-    return 1;    # success
+    # Now we enable the account
+    return system("$smbPasswdProg -e $username >/dev/null") ? 0 : 1;
 }
 
 =pod
