diff -Nur -x '*.orig' -x '*.rej' e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/config/utf8.pm mezzanine_patched_e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/config/utf8.pm
--- e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/config/utf8.pm	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/config/utf8.pm	2008-07-31 09:21:16.000000000 -0600
@@ -0,0 +1,25 @@
+#----------------------------------------------------------------------
+# Copyright 1999-2008 Mitel Networks Corporation
+# This program is free software; you can redistribute it and/or
+# modify it under the same terms as Perl itself.
+#----------------------------------------------------------------------
+
+package esmith::config::utf8;
+use warnings;
+use strict;
+
+use vars qw(@ISA);
+@ISA = qw(esmith::config);
+
+sub _read_binmode
+{
+    return ":encoding(UTF-8)";
+}
+
+sub _write_binmode
+{
+    return ":utf8";
+}
+
+1;
+
diff -Nur -x '*.orig' -x '*.rej' e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/ConfigDB/UTF8.pm mezzanine_patched_e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/ConfigDB/UTF8.pm
--- e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/ConfigDB/UTF8.pm	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/ConfigDB/UTF8.pm	2008-07-31 09:20:19.000000000 -0600
@@ -0,0 +1,22 @@
+#----------------------------------------------------------------------
+# Copyright 1999-2008 Mitel Networks Corporation
+# This program is free software; you can redistribute it and/or
+# modify it under the same terms as Perl itself.
+#----------------------------------------------------------------------
+
+package esmith::ConfigDB::UTF8;
+
+use strict;
+use warnings;
+
+use esmith::DB::db;
+use esmith::config::utf8;
+our @ISA = qw( esmith::DB::db );
+
+sub tie_class
+{
+    return 'esmith::config::utf8';
+}
+
+1;
+
diff -Nur -x '*.orig' -x '*.rej' e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/NavigationDB.pm mezzanine_patched_e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/NavigationDB.pm
--- e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/NavigationDB.pm	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/NavigationDB.pm	2008-07-31 09:19:13.000000000 -0600
@@ -0,0 +1,16 @@
+#----------------------------------------------------------------------
+# Copyright 1999-2008 Mitel Networks Corporation
+# This program is free software; you can redistribute it and/or
+# modify it under the same terms as Perl itself.
+#----------------------------------------------------------------------
+
+package esmith::NavigationDB;
+
+use strict;
+use warnings;
+
+use esmith::ConfigDB::UTF8;
+our @ISA = qw( esmith::ConfigDB::UTF8 );
+
+1;
+
