diff -Nur -x '*.orig' -x '*.rej' e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/util.pm mezzanine_patched_e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/util.pm
--- e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/util.pm	2008-08-09 18:52:53.000000000 -0600
+++ mezzanine_patched_e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/util.pm	2008-08-09 18:52:39.000000000 -0600
@@ -840,19 +840,19 @@
 
     use MIME::Base64 qw(encode_base64);
 
-    unless ( open( RANDOM, "/dev/random" ) )
+    unless ( open( RANDOM, "/dev/urandom" ) )
     {
-        warn "Could not open /dev/random: $!";
+        warn "Could not open /dev/urandom: $!";
         return undef;
     }
 
     my $buf = "not set";
 
     # 57 bytes is a full line of Base64 coding, and contains
-    # 456 bits of randomness - given a perfectly random /dev/random
+    # 456 bits of randomness - given a perfectly random /dev/urandom
     if ( read( RANDOM, $buf, 57 ) != 57 )
     {
-        warn("Short read from /dev/random: $!");
+        warn("Short read from /dev/urandom: $!");
         return undef;
     }
     close RANDOM;
