diff -Nur -x '*.orig' -x '*.rej' e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/console.pm mezzanine_patched_e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/console.pm
--- e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/console.pm	2007-02-22 17:53:21.000000000 -0500
+++ mezzanine_patched_e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/console.pm	2007-02-22 17:52:39.000000000 -0500
@@ -285,12 +285,12 @@
     my $left  = defined $params{left} ? $params{left} : gettext("Back");
     my $right = defined $params{right} ? $params{right} : gettext("Next");
 
-    $self->whiptail ("--title",  $title,
+    $self->dialog ("--title",  $title,
                    "--clear",
-                   "--password",
+                   "--insecure",
                    "--cancel-label",  $left,
                    "--ok-label", $right,
-                   "--inputbox", "\n" . $message_box,
+                   "--passwordbox", "\n" . $message_box,
                    SCREEN_ROWS,
                    SCREEN_COLUMNS,
                   );
