diff -Nur -x '*.orig' -x '*.rej' e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/util.pm mezzanine_patched_e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/util.pm
--- e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/util.pm	2007-01-26 18:51:29.000000000 -0700
+++ mezzanine_patched_e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/util.pm	2007-05-21 21:19:55.000000000 -0600
@@ -290,7 +290,7 @@
 
     # first, calculate the prefix (/??) given the netmask
     my $len = 0;
-    for ( my $bits = $netmaskBits ; $bits ; $bits <<= 1 )
+    for ( my $bits = $netmaskBits ; $bits & 0xFFFFFFFF ; $bits <<= 1 )
     {
         $len++;
     }
