diff -Nur -x '*.orig' -x '*.rej' e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/I18N.pm mezzanine_patched_e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/I18N.pm
--- e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/I18N.pm	2008-04-02 14:09:35.000000000 -0600
+++ mezzanine_patched_e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/I18N.pm	2008-04-02 14:09:16.000000000 -0600
@@ -105,7 +105,7 @@
 
 sub fallbackLocale()
 {
-    return "en_US";
+    return "en_US.utf8";
 }
 
 =head2 preferredLocale()
@@ -154,8 +154,9 @@
     my $locale;
 
     $locale = $self->langtag2locale($opt) if ($opt);
-
     $locale ||= $self->preferredLocale;
+    $locale =~ s{(?:\..*)?$}{.utf8};
+    $locale =~ s{.utf8$}{} if $text_domain eq 'server-console';
 
     $ENV{'LANGUAGE'} = $locale;
     $ENV{'LANG'}     = $ENV{'LANGUAGE'};
@@ -207,11 +208,19 @@
 
 	unless (-d "/usr/lib/locale/$locale")
 	{
+	    $locale = "da_DK" if ($opt =~ /^da(-.*)?/);
+	    $locale = "de_DE" if ($opt =~ /^de(-.*)?/);
+	    $locale = "el_GR" if ($opt =~ /^el(-.*)?/);
 	    $locale = "en_US" if ($opt =~ /^en(-.*)?/);
 	    $locale = "es_ES" if ($opt =~ /^es(-.*)?/);
 	    $locale = "fr_CA" if ($opt =~ /^fr(-.*)?/);
-	    $locale = "de_DE" if ($opt =~ /^de(-.*)?/);
+	    $locale = "hu_HU" if ($opt =~ /^hu(-.*)?/);
+	    $locale = "id_ID" if ($opt =~ /^id(-.*)?/);
 	    $locale = "it_IT" if ($opt =~ /^it(-.*)?/);
+	    $locale = "nl_NL" if ($opt =~ /^nl(-.*)?/);
+	    $locale = "pt_BR" if ($opt =~ /^pt(-.*)?/);
+	    $locale = "sl_SL" if ($opt =~ /^sl(-.*)?/);
+	    $locale = "sv_SE" if ($opt =~ /^sv(-.*)?/);
 	}
     }
     else 
