diff -Nur -x '*.orig' -x '*.rej' e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/util.pm mezzanine_patched_e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/util.pm
--- e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/util.pm	2008-01-07 06:26:45.000000000 -0700
+++ mezzanine_patched_e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/util.pm	2008-01-07 06:25:43.000000000 -0700
@@ -574,7 +574,7 @@
     $reason = 'it does not contain uppercase characters' if (not $password =~ /[A-Z]/);
     $reason = 'it does not contain lowercase characters' if (not $password =~ /[a-z]/);
     $reason = 'it does not contain special characters' if (not $password =~ /\W|_/);
-    return $reason unless ($reason eq 'ok' || $strength eq 'strong');
+    return $reason if ($reason ne 'ok' && $strength eq 'strong');
 
     $reason = fascist_check($password, '/usr/lib/cracklib_dict');
     $reason ||= 'the password check failed';
