diff -Nur -x '*.orig' -x '*.rej' e-smith-lib-1.15.3/root/usr/lib/perl5/site_perl/esmith/util.pm mezzanine_patched_e-smith-lib-1.15.3/root/usr/lib/perl5/site_perl/esmith/util.pm
--- e-smith-lib-1.15.3/root/usr/lib/perl5/site_perl/esmith/util.pm	2005-12-25 14:40:21.397462652 +1100
+++ mezzanine_patched_e-smith-lib-1.15.3/root/usr/lib/perl5/site_perl/esmith/util.pm	2005-12-25 14:39:54.273181972 +1100
@@ -1181,16 +1181,17 @@
             if (-l "${old_dbhome}/$file")
             {
                 warn "symlink called ${old_dbhome}/$file exists\n";
+		next;
             }
-            elsif (-s "${dbhome}/$file")
+
+            if (-s "${dbhome}/$file")
             {
                 warn "${old_dbhome}/$file and ${dbhome}/$file exist\n";
+                rename "${dbhome}/$file", "${dbhome}/$file." . time;
             }
-            else
-            {
-                warn "Rename ${old_dbhome}/$file => ${dbhome}/$file\n";
-                rename "${old_dbhome}/$file", "${dbhome}/$file";
-            }
+
+	    warn "Rename ${old_dbhome}/$file => ${dbhome}/$file\n";
+	    rename "${old_dbhome}/$file", "${dbhome}/$file";
         }
     }
 
