diff -Nur -x '*.orig' -x '*.rej' e-smith-lib-1.16.0/root/usr/lib/perl5/site_perl/esmith/util.pm mezzanine_patched_e-smith-lib-1.16.0/root/usr/lib/perl5/site_perl/esmith/util.pm
--- e-smith-lib-1.16.0/root/usr/lib/perl5/site_perl/esmith/util.pm	2006-12-12 07:51:11.000000000 -0700
+++ mezzanine_patched_e-smith-lib-1.16.0/root/usr/lib/perl5/site_perl/esmith/util.pm	2006-12-12 07:50:49.000000000 -0700
@@ -553,6 +553,39 @@
 change passwords for a single underlying password database,
 for example /etc/passwd, /etc/samba/smbpasswd, etc.
 
+=head2 validatePassword($password, $strength)
+
+Validate Unix password.
+
+=cut
+
+sub validatePassword($$)
+{
+    my ( $password, $strength ) = @_;
+    use Crypt::Cracklib;
+
+    $strength ||= 'normal';
+    return 'ok' if ($strength eq 'none');
+
+    my $reason = 'ok';
+    $reason = 'it does not contain numbers' if (not $password =~ /\d/);
+    $reason = 'it does not contain uppercase characters' if (not $password =~ /[A-Z]/);
+    $reason = 'it does not contain lowercase characters' if (not $password =~ /[a-z]/);
+    $reason = 'it does not contain special characters' if (not $password =~ /\W|_/);
+    $reason = 'it is too short' if (not length($password) > 6);
+
+    return $reason unless ($reason eq 'ok');
+    return 'ok' unless ($strength eq "strong");
+
+    $reason = fascist_check($password, '/usr/lib/cracklib_dict');
+    $reason ||= 'the password check failed';
+
+    return 'ok' if (lc($reason) eq 'ok');
+    return $reason;
+}
+
+=pod
+
 =head2 setUnixPassword($username, $password)
 
 Set Unix password
