diff -Nur -x '*.orig' -x '*.rej' e-smith-lib-1.15.3/root/usr/lib/perl5/site_perl/esmith/config.pm mezzanine_patched_e-smith-lib-1.15.3/root/usr/lib/perl5/site_perl/esmith/config.pm
--- e-smith-lib-1.15.3/root/usr/lib/perl5/site_perl/esmith/config.pm	2005-12-27 09:44:37.482446243 +1100
+++ mezzanine_patched_e-smith-lib-1.15.3/root/usr/lib/perl5/site_perl/esmith/config.pm	2005-12-27 09:44:18.489750759 +1100
@@ -371,10 +371,8 @@
 
     if ($filename =~ m:^/home/e-smith/\w+$: )
     {
-        cluck "*WARNING* esmith::config($filename) called with old \n" .
-              "database path. Direct use of esmith::config has been\n" .
-	      "deprecated since release 5.5\n" .
-              "The following package needs to be updated:\n";
+        cluck "*WARNING* esmith::config($filename) called with old " .
+              "database path. The following package needs to be updated: ";
 
         $filename =~ s:e-smith:e-smith/db:;
     }
