diff -Nur -x '*.orig' -x '*.rej' e-smith-lib-1.16.0/root/usr/lib/perl5/site_perl/esmith/ethernet.pm mezzanine_patched_e-smith-lib-1.16.0/root/usr/lib/perl5/site_perl/esmith/ethernet.pm
--- e-smith-lib-1.16.0/root/usr/lib/perl5/site_perl/esmith/ethernet.pm	2006-02-16 22:52:14.000000000 -0500
+++ mezzanine_patched_e-smith-lib-1.16.0/root/usr/lib/perl5/site_perl/esmith/ethernet.pm	2006-04-13 11:07:18.000000000 -0400
@@ -237,9 +237,9 @@
 {
     my @devs;
     my $r;
-    unless (open(HWC, "/etc/sysconfig/hwconf"))
+    unless (open(HWC, "/usr/sbin/kudzu --probe --class network|"))
     {
-	warn "Could not open /etc/sysconfig/hwconf: $!";
+	warn "Could not run kudzu probe: $!";
 	return "";
     }
     while (<HWC>)
