diff -Nur -x '*.orig' -x '*.rej' e-smith-lib-1.15.3/root/usr/lib/perl5/site_perl/esmith/ethernet.pm mezzanine_patched_e-smith-lib-1.15.3/root/usr/lib/perl5/site_perl/esmith/ethernet.pm
--- e-smith-lib-1.15.3/root/usr/lib/perl5/site_perl/esmith/ethernet.pm	2006-01-19 18:00:29.000000000 -0500
+++ mezzanine_patched_e-smith-lib-1.15.3/root/usr/lib/perl5/site_perl/esmith/ethernet.pm	2006-01-19 17:59:53.000000000 -0500
@@ -237,8 +237,6 @@
 {
     my @devs;
     my $r;
-    # Ensure that hwconf file is up to date
-    system("/usr/sbin/kudzu", "-q", "--class=network");
     unless (open(HWC, "/etc/sysconfig/hwconf"))
     {
 	warn "Could not open /etc/sysconfig/hwconf: $!";
