diff -Nur -x '*.orig' -x '*.rej' e-smith-lib-1.15.3/root/etc/e-smith/events/actions/adjust-services mezzanine_patched_e-smith-lib-1.15.3/root/etc/e-smith/events/actions/adjust-services
--- e-smith-lib-1.15.3/root/etc/e-smith/events/actions/adjust-services	2005-11-19 07:49:22.443855994 +1100
+++ mezzanine_patched_e-smith-lib-1.15.3/root/etc/e-smith/events/actions/adjust-services	2005-11-19 07:49:00.304734575 +1100
@@ -121,7 +121,7 @@
 		@actions,
 		# bring the service up if it is enabled (and we're not
 		# stopping it or running it once)
-		($enabled && (grep { /^(down|stop|d|once|o)$/ } @actions) ? () : 'up'),
+		((not $enabled) || (grep { /^(down|stop|d|once|o)$/ } @actions) ? () : 'up'),
 	    );
     }
     else
