diff -Nur -x '*.orig' -x '*.rej' e-smith-lib-1.15.3/root/usr/lib/perl5/site_perl/esmith/NetworksDB.pm mezzanine_patched_e-smith-lib-1.15.3/root/usr/lib/perl5/site_perl/esmith/NetworksDB.pm
--- e-smith-lib-1.15.3/root/usr/lib/perl5/site_perl/esmith/NetworksDB.pm	2006-01-10 10:06:48.216266820 -0500
+++ mezzanine_patched_e-smith-lib-1.15.3/root/usr/lib/perl5/site_perl/esmith/NetworksDB.pm	2006-01-10 10:04:25.201306905 -0500
@@ -101,10 +101,12 @@
 
 Note: The elements are space separated, which is suitable for use in
 hosts.allow, smb.conf and httpd.conf. httpd.conf does not permit 
-comma separated lists in C<allow from> directives.
+comma separated lists in C<allow from> directives. Each element is either
+an IP address, or a network/netmask string.
 
-If called in list context, returns the array of network/netmask strings.
-It's trivial, of course, to convert an array to a comma separated list :-)
+If called in list context, returns the array of addresses and network/netmask
+strings. It's trivial, of course, to convert an array to a comma separated
+list :-)
 
 =cut
 
@@ -123,8 +125,10 @@
     {
         foreach my $network ( $self->networks )
         {
+	    my $element = $network->key;
             my $mask = $network->prop('Mask');
-            push @localAccess, $network->key . "/$mask";
+	    $element .= "/$mask" unless ($mask eq "255.255.255.255");
+            push @localAccess, $element;
         }
     }
     elsif ( $access eq "public" )
