diff -Nur -x '*.orig' -x '*.rej' e-smith-lib-1.15.2/root/etc/e-smith/tests/10e-smith-lib/configuration.conf mezzanine_patched_e-smith-lib-1.15.2/root/etc/e-smith/tests/10e-smith-lib/configuration.conf
--- e-smith-lib-1.15.2/root/etc/e-smith/tests/10e-smith-lib/configuration.conf	2003-05-28 15:36:04.000000000 -0600
+++ mezzanine_patched_e-smith-lib-1.15.2/root/etc/e-smith/tests/10e-smith-lib/configuration.conf	2005-07-19 16:02:14.068736320 -0600
@@ -36,7 +36,7 @@
 LocalNetmask=255.255.255.0
 MinUid=5000
 PasswordSet=yes
-PreviousConfiguration=/home/e-smith/configuration.previous
+PreviousConfiguration=/home/e-smith/db/configuration.previous
 SMTPSmartHost=
 SambaDomainMaster=no
 SambaServerName=pretz
diff -Nur -x '*.orig' -x '*.rej' e-smith-lib-1.15.2/root/usr/lib/perl5/site_perl/esmith/config.pm mezzanine_patched_e-smith-lib-1.15.2/root/usr/lib/perl5/site_perl/esmith/config.pm
--- e-smith-lib-1.15.2/root/usr/lib/perl5/site_perl/esmith/config.pm	2005-07-15 15:32:12.000000000 -0600
+++ mezzanine_patched_e-smith-lib-1.15.2/root/usr/lib/perl5/site_perl/esmith/config.pm	2005-07-19 16:02:14.075735256 -0600
@@ -12,7 +12,7 @@
 use Sys::Syslog qw(:DEFAULT setlogsock);
 use Fcntl qw(:DEFAULT :flock);
 
-my $Default_Config = '/home/e-smith/configuration';
+my $Default_Config = '/home/e-smith/db/configuration';
 
 =pod
 
@@ -70,7 +70,7 @@
 deletes are immediately written back to the $config_file.
 
 If no $config_file is supplied it falls back to the environment variable 
-ESMITH_CONFIG_DB, and finally defaults to F</home/e-smith/configuration>
+ESMITH_CONFIG_DB, and finally defaults to F</home/e-smith/db/configuration>
 
 If the $config_file doesn't exist it will create one for you.
 
@@ -360,7 +360,7 @@
 
 #------------------------------------------------------------
 # Constructor for the tied hash. If filename not specified,
-# defaults to '/home/e-smith/configuration'.
+# defaults to '/home/e-smith/db/configuration'.
 #------------------------------------------------------------
 
 sub TIEHASH
diff -Nur -x '*.orig' -x '*.rej' e-smith-lib-1.15.2/root/usr/lib/perl5/site_perl/esmith/DB/db.pm mezzanine_patched_e-smith-lib-1.15.2/root/usr/lib/perl5/site_perl/esmith/DB/db.pm
--- e-smith-lib-1.15.2/root/usr/lib/perl5/site_perl/esmith/DB/db.pm	2005-02-22 10:06:01.000000000 -0700
+++ mezzanine_patched_e-smith-lib-1.15.2/root/usr/lib/perl5/site_perl/esmith/DB/db.pm	2005-07-19 16:02:14.077734952 -0600
@@ -78,6 +78,7 @@
 sub create
 {
     my ( $class, $file ) = @_;
+    $file = $class->_file_path($file);
     my $self;
 
     eval {
@@ -131,6 +132,7 @@
 sub open
 {
     my ( $class, $file ) = @_;
+    $file = $class->_file_path($file);
     my $self = $class->_init($file);
 
     if ( -e $file && !-w $file )
@@ -155,6 +157,7 @@
 sub open_ro
 {
     my ( $class, $file ) = @_;
+    $file = $class->_file_path($file);
     my $self = $class->_init($file);
 
     $self->{ro} = 1;
@@ -210,6 +213,27 @@
     return $self;
 }
 
+sub _file_path
+{
+    my ( $class, $file ) = @_;
+
+    if ($file =~ m:/:)
+    {
+	warn "Pathname passed to _file_path($file)";
+	return $file;
+    }
+
+    if (-e "/home/e-smith/db/$file")
+    {
+	return "/home/e-smith/db/$file";
+    } elsif (-e "/home/e-smith/$file") {
+	warn "Database found in old location /home/e-smith/$file";
+	return "/home/e-smith/$file";
+    } else {
+	return "/home/e-smith/db/$file";
+    }
+}
+
 =item B<as_hash>
 
 =begin testing
diff -Nur -x '*.orig' -x '*.rej' e-smith-lib-1.15.2/root/usr/lib/perl5/site_perl/esmith/db.pm mezzanine_patched_e-smith-lib-1.15.2/root/usr/lib/perl5/site_perl/esmith/db.pm
--- e-smith-lib-1.15.2/root/usr/lib/perl5/site_perl/esmith/db.pm	2005-07-19 16:08:25.195684680 -0600
+++ mezzanine_patched_e-smith-lib-1.15.2/root/usr/lib/perl5/site_perl/esmith/db.pm	2005-07-19 16:02:14.079734648 -0600
@@ -701,9 +701,22 @@
 
 sub _db_path($)
 {
-    my ($arg) = @_;
+    my ($file) = @_;
 
-    return ($arg =~ m:^/:) ? $arg : "/home/e-smith/${arg}";
+    if ($file =~ m:^/:)
+    {
+	warn "Pathname passed to _db_path($file)";
+	return $file;
+    }
+    if (-e "/home/e-smith/db/$file")
+    {
+	return "/home/e-smith/db/$file";
+    } elsif (-e "/home/e-smith/$file") {
+	warn "Database found in old location /home/e-smith/$file";
+	return "/home/e-smith/$file";
+    } else {
+	return "/home/e-smith/db/$file";
+    }
 }
 
 =back
diff -Nur -x '*.orig' -x '*.rej' e-smith-lib-1.15.2/root/usr/lib/perl5/site_perl/esmith/DB.pm mezzanine_patched_e-smith-lib-1.15.2/root/usr/lib/perl5/site_perl/esmith/DB.pm
--- e-smith-lib-1.15.2/root/usr/lib/perl5/site_perl/esmith/DB.pm	2005-05-05 07:55:42.000000000 -0600
+++ mezzanine_patched_e-smith-lib-1.15.2/root/usr/lib/perl5/site_perl/esmith/DB.pm	2005-07-19 16:02:14.081734344 -0600
@@ -453,7 +453,7 @@
         {
             processTemplate(
                 {
-                    MORE_DATA             => { 'DB_FILENAME' => $self->{file} },
+                    MORE_DATA             => { 'DB_FILENAME' => $dbfile },
                     TEMPLATE_PATH         => '',
                     OUTPUT_TYPE           => 'string',
                     TEMPLATE_EXPAND_QUEUE =>
diff -Nur -x '*.orig' -x '*.rej' e-smith-lib-1.15.2/root/usr/lib/perl5/site_perl/esmith/util.pm mezzanine_patched_e-smith-lib-1.15.2/root/usr/lib/perl5/site_perl/esmith/util.pm
--- e-smith-lib-1.15.2/root/usr/lib/perl5/site_perl/esmith/util.pm	2005-06-21 13:51:19.000000000 -0600
+++ mezzanine_patched_e-smith-lib-1.15.2/root/usr/lib/perl5/site_perl/esmith/util.pm	2005-07-19 16:02:14.094732368 -0600
@@ -1272,11 +1272,14 @@
     # Optionally take an argument to the db root, for testing purposes.
     my %defaults = (
         dbroot => '/etc/e-smith/db',
-        dbhome => '/home/e-smith'
+        dbhome => '/home/e-smith/db',
+        old_dbhome => '/home/e-smith',
     );
     my %args   = ( %defaults, @_ );
     my $dbroot = $args{dbroot};
     my $dbhome = $args{dbhome};
+    my $old_dbhome = $args{old_dbhome};
+
     local *DH;
     opendir DH, $dbroot
       or die "Could not open $dbroot: $!";
@@ -1299,7 +1302,20 @@
 
         eval
         {
-            my $h = esmith::ConfigDB->open("$dbhome/$file");
+            if (-f "${old_dbhome}/$file")
+            {
+                if (-s "${dbhome}/$file")
+                {
+                    warn "${old_dbhome}/$file and ${dbhome}/$file exist\n";
+                }
+                else
+                {
+                    warn "Rename ${old_dbhome}/$file => ${dbhome}/$file\n";
+                    rename "${old_dbhome}/$file", "${dbhome}/$file";
+                }
+            }
+
+            my $h = esmith::ConfigDB->open($file);
             if ($h)
             {
                 warn "Migrating existing database $file\n";
@@ -1317,7 +1333,7 @@
                 warn "Creating database $file and setting defaults\n";
 
                 # create() and load defaults
-                unless ( $h = esmith::ConfigDB->create("$dbhome/$file") )
+                unless ( $h = esmith::ConfigDB->create($file) )
                 {
                     warn "Could not create $file db: " . esmith::DB->error;
                 }
