diff -Nur -x '*.orig' -x '*.rej' e-smith-lib-1.15.1/root/usr/lib/perl5/site_perl/esmith/DB/db.pm mezzanine_patched_e-smith-lib-1.15.1/root/usr/lib/perl5/site_perl/esmith/DB/db.pm
--- e-smith-lib-1.15.1/root/usr/lib/perl5/site_perl/esmith/DB/db.pm	2005-07-18 01:26:13.574678920 -0600
+++ mezzanine_patched_e-smith-lib-1.15.1/root/usr/lib/perl5/site_perl/esmith/DB/db.pm	2005-07-18 01:25:12.566759008 -0600
@@ -78,6 +78,7 @@
 sub create
 {
     my ( $class, $file ) = @_;
+    $file = $class->_file_path($file);
     my $self;
 
     eval {
@@ -131,6 +132,7 @@
 sub open
 {
     my ( $class, $file ) = @_;
+    $file = $class->_file_path($file);
     my $self = $class->_init($file);
 
     if ( -e $file && !-w $file )
@@ -155,6 +157,7 @@
 sub open_ro
 {
     my ( $class, $file ) = @_;
+    $file = $class->_file_path($file);
     my $self = $class->_init($file);
 
     $self->{ro} = 1;
@@ -210,6 +213,25 @@
     return $self;
 }
 
+sub _file_path
+{
+    my ( $class, $file ) = @_;
+
+    if ($file =~ m:^/:)
+    {
+	warn "Pathname passed to _file_path($file)";
+	return $file;
+    }
+
+    if (-e "/home/e-smith/db/$file")
+    {
+	return "/home/e-smith/db/$file" if (-e "/home/e-smith/db/$file");
+    } else {
+	warn "Database not found in new location /home/e-smith/db/$file";
+	return "/home/e-smith/$file";
+    }
+}
+
 =item B<as_hash>
 
 =begin testing
diff -Nur -x '*.orig' -x '*.rej' e-smith-lib-1.15.1/root/usr/lib/perl5/site_perl/esmith/db.pm mezzanine_patched_e-smith-lib-1.15.1/root/usr/lib/perl5/site_perl/esmith/db.pm
--- e-smith-lib-1.15.1/root/usr/lib/perl5/site_perl/esmith/db.pm	2005-07-18 01:26:14.582532856 -0600
+++ mezzanine_patched_e-smith-lib-1.15.1/root/usr/lib/perl5/site_perl/esmith/db.pm	2005-07-18 01:22:53.713489424 -0600
@@ -701,9 +701,21 @@
 
 sub _db_path($)
 {
-    my ($arg) = @_;
+    my ($file) = @_;
 
-    return ($arg =~ m:^/:) ? $arg : "/home/e-smith/db/${arg}";
+    if ($file =~ m:^/:)
+    {
+	warn "Pathname passed to _db_path($file)";
+	return $file;
+    }
+
+    if (-e "/home/e-smith/db/$file")
+    {
+	return "/home/e-smith/db/$file" if (-e "/home/e-smith/db/$file");
+    } else {
+	warn "Database not found in new location /home/e-smith/db/$file";
+	return "/home/e-smith/$file";
+    }
 }
 
 =back
