diff -Nur -x '*.orig' -x '*.rej' e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/console.pm mezzanine_patched_e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/console.pm
--- e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/console.pm	2007-01-26 18:51:29.000000000 -0700
+++ mezzanine_patched_e-smith-lib-1.18.0/root/usr/lib/perl5/site_perl/esmith/console.pm	2007-01-28 09:47:49.000000000 -0700
@@ -205,7 +205,7 @@
 
     my $db = esmith::ConfigDB->open_ro or die "Couldn't open ConfigDB\n";
 
-    sprintf("%-35s%45s",
+    sprintf("%-33s%45s",
     ($db->get_prop('sysconfig', 'ProductName') || "SME Server") . " " .
     ($db->get_prop('sysconfig', 'ReleaseVersion') || "UNKNOWN"),
     "Copyright (C) 1999-2006 Mitel Corporation"
