diff -Nur -x '*.orig' -x '*.rej' e-smith-lib-1.15.3/root/usr/lib/perl5/site_perl/esmith/AccountsDB.pm mezzanine_patched_e-smith-lib-1.15.3/root/usr/lib/perl5/site_perl/esmith/AccountsDB.pm
--- e-smith-lib-1.15.3/root/usr/lib/perl5/site_perl/esmith/AccountsDB.pm	2005-07-20 03:13:46.000000000 +1000
+++ mezzanine_patched_e-smith-lib-1.15.3/root/usr/lib/perl5/site_perl/esmith/AccountsDB.pm	2006-01-09 14:56:52.372607372 +1100
@@ -515,7 +515,7 @@
         $self = shift;
     }
     my $name = shift;
-    return ($name =~ /[^0-9a-z\-]/ or $name !~ /^[a-z]/) ? undef : 'OK';
+    return ($name =~ /[^0-9a-z\-_\.]/ or $name !~ /^[a-z]/) ? undef : 'OK';
 }
 
 =head1 AUTHOR
